File: examples/biased-coin-flip/prior-pdf.py
from sys import argv, stdin, stdout, stderr

# Process arguments
if len(argv) != 3:
    stderr.write("Usage: {} Q_LOW Q_HIGH\n"
            "Return probability density of uniform distribution with \n"
            "lower bound Q_LOW and upper bound Q_HIGH "
            "for given parameter\n".format(argv[0]))

    exit(1)

q_low = float(argv[1])
q_high = float(argv[2])

# Read parameter from stdin
stderr.write("Enter q\n")
q = float(stdin.readline())

# Check if q is within bounds
if q_low <= q and q <= q_high:
    density = 1.0 / (q_high - q_low)
    stdout.write("{}\n".format(density))
else:
    stdout.write("0\n")

File: examples/biased-coin-flip/prior-pdf.cc
#include <string>
#include <iostream>

int main(int argc, char *argv[])
{
    // Process arguments
    if (argc != 3)
    {
        std::cerr << "Usage: " << argv[0] << " Q_LOW Q_HIGH\n"
            "Return probability density of uniform distribution with \n"
            "lower bound Q_LOW and upper bound Q_HIGH "
            "for given parameter\n";
        return 1;
    }

    double q_low = std::stod(argv[1]);
    double q_high = std::stod(argv[2]);

    if (q_high <= q_low)
    {
        std::cerr << "Error: Q_LOW must be strictly less than Q_HIGH\n";
        return 1;
    }

    // Read parameter from stdin
    double q;
    std::cin >> q;

    if (std::cin.fail())
    {
        std::cerr << "Error: could not read q from stdin\n";
        return 1;
    }

    // Check if q is within bounds
    std::cout.precision(17);
    if ( (q_low <= q) && (q <= q_high) )
    {
        std::cout << 1.0 / (q_high - q_low) << std::endl;
    }
    else
    {
        std::cout << 0 << std::endl;
    }

    return 0;
}
